### ISSUE TITLE
[BUG] Test resources are not made available when the test container is build

### ISSUE THREAD

**Issue by DriesSchaumont** [viash-io/viash#651]
### What happened?

When adding a resource to `test_resources`, they are not available when building the container where the test script is executed in.

### Steps to reproduce

```
mkdir -p /tmp/test_bug
cd /tmp/test_bug
cat > _viash.yaml << 'HERE'
viash_version: 0.8.4

source: src
target: target
HERE

mkdir -p src/dummycomponent
touch src/dummycomponent/a_test_resource.txt

cat > src/dummycomponent/config.vsh.yaml << 'HERE'
functionality:
  name: myscript
  namespace: "dummycomponent"
  resources:
    - type: bash_script
      path: script.sh
      text: |
        ls .
  test_resources:
    - type: bash_script
      path: test.sh
      text: |
        echo "OK!" && exit 0
    - path: a_test_resource.txt
platforms:
  - type: docker
    image: ubuntu:latest
    setup:
      - type: docker
        run: |
          ls -latr .
    test_setup:
      - type: docker
        run: |
          [ -f "a_test_resource.txt" ] || exit 8
  - type: native
  - type: nextflow
HERE

viash test ./src/dummycomponent/config.vsh.yaml -k true --setup build
```

### Expected behavior

The test resource can be used using `test_setup`

### Relevant log output

```
viash test ./src/dummycomponent/config.vsh.yaml -k true --setup build
Running tests in temporary directory: '/tmp/viash_test_myscript11135655716862917029'
====================================================================
+/tmp/viash_test_myscript11135655716862917029/build_executable/myscript ---verbosity 6 ---setup build
[notice] Building container 'dummycomponent/myscript:test' with Dockerfile
[info] Running 'docker build -t dummycomponent/myscript:test --no-cache /tmp/viash_test_myscript11135655716862917029/build_executable -f /tmp/viash_test_myscript11135655716862917029/build_executable/tmp/dockerbuild-myscript-jfwfov/Dockerfile'
#0 building with "default" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 361B done
#1 DONE 0.1s

#2 [internal] load metadata for docker.io/library/ubuntu:latest
#2 DONE 0.4s

#3 [internal] load .dockerignore
#3 transferring context: 2B done
#3 DONE 0.0s

#4 [1/3] FROM docker.io/library/ubuntu:latest@sha256:f9d633ff6640178c2d0525017174a688e2c1aef28f0a0130b26bd5554491f0da
#4 CACHED

#5 [2/3] RUN ls -latr .
#5 0.381 total 17
#5 0.381 drwxr-xr-x   2 root root   2 Apr 18  2022 home
#5 0.381 drwxr-xr-x   2 root root   2 Apr 18  2022 boot
#5 0.381 lrwxrwxrwx   1 root root   8 Feb 12 14:02 sbin -> usr/sbin
#5 0.381 lrwxrwxrwx   1 root root  10 Feb 12 14:02 libx32 -> usr/libx32
#5 0.381 lrwxrwxrwx   1 root root   9 Feb 12 14:02 lib64 -> usr/lib64
#5 0.381 lrwxrwxrwx   1 root root   9 Feb 12 14:02 lib32 -> usr/lib32
#5 0.381 lrwxrwxrwx   1 root root   7 Feb 12 14:02 lib -> usr/lib
#5 0.381 lrwxrwxrwx   1 root root   7 Feb 12 14:02 bin -> usr/bin
#5 0.381 drwxr-xr-x  14 root root  14 Feb 12 14:02 usr
#5 0.381 drwxr-xr-x   2 root root   2 Feb 12 14:02 srv
#5 0.381 drwxr-xr-x   2 root root   2 Feb 12 14:02 opt
#5 0.381 drwxr-xr-x   2 root root   2 Feb 12 14:02 mnt
#5 0.381 drwxr-xr-x   2 root root   2 Feb 12 14:02 media
#5 0.381 drwxr-xr-x  11 root root  13 Feb 12 14:06 var
#5 0.381 drwx------   2 root root   4 Feb 12 14:06 root
#5 0.381 drwxrwxrwt   2 root root   2 Feb 12 14:06 tmp
#5 0.381 drwxr-xr-x   5 root root   5 Feb 12 14:06 run
#5 0.381 drwxr-xr-x  32 root root  73 Feb 12 14:06 etc
#5 0.381 drwxr-xr-x  17 root root  23 Feb 16 09:55 ..
#5 0.381 drwxr-xr-x  17 root root  23 Feb 16 09:55 .
#5 0.381 dr-xr-xr-x  13 root root   0 Feb 16 10:19 sys
#5 0.381 dr-xr-xr-x 790 root root   0 Feb 16 10:19 proc
#5 0.381 drwxr-xr-x   5 root root 340 Feb 16 10:19 dev
#5 DONE 0.4s

#6 [3/3] RUN [ -f "a_test_resource.txt" ] || exit 8
#6 ERROR: process "/bin/sh -c [ -f \"a_test_resource.txt\" ] || exit 8" did not complete successfully: exit code: 8
------
 > [3/3] RUN [ -f "a_test_resource.txt" ] || exit 8:
------
Dockerfile:8
--------------------
   6 |     RUN ls -latr .
   7 |
   8 | >>> RUN [ -f "a_test_resource.txt" ] || exit 8
   9 |
  10 |     LABEL org.opencontainers.image.description="Companion container for running component dummycomponent myscript"
--------------------
ERROR: failed to solve: process "/bin/sh -c [ -f \"a_test_resource.txt\" ] || exit 8" did not complete successfully: exit code: 8
[error] Error occurred while building container 'dummycomponent/myscript:test'
ERROR! Setup failed!
Unexpected error occurred! If you think this is a bug, please post
create an issue at https://github.com/viash-io/viash/issues containing
a reproducible example and the stack trace below.

viash - 0.8.4
Stacktrace:
java.lang.RuntimeException: Setup failed!
	at io.viash.ViashTest$.apply(ViashTest.scala:134)
	at io.viash.Main$.mainCLI(Main.scala:253)
	at io.viash.Main$.mainCLIOrVersioned(Main.scala:123)
	at io.viash.Main$.main(Main.scala:58)
	at io.viash.Main.main(Main.scala)
```

### Version

- OS: ubuntu 22.04
- Viash 0.8.4

### Confirmation

- [X] I have searched the existing issues to make sure this is not a duplicate.
- [X] I have provided clear and concise information about the bug.

**Comment by DriesSchaumont** [comment]
I just noticed that my example does not cover the issue. This should more clearly describe what is going on:

```
mkdir -p /tmp/test_bug
cd /tmp/test_bug
cat > _viash.yaml << 'HERE'
viash_version: 0.8.4

source: src
target: target
HERE

mkdir -p src/dummycomponent
touch src/dummycomponent/a_test_resource.txt

cat > src/dummycomponent/config.vsh.yaml << 'HERE'
functionality:
  name: myscript
  namespace: "dummycomponent"
  resources:
    - type: bash_script
      path: script.sh
      text: |
        ls .
  test_resources:
    - type: bash_script
      path: test.sh
      text: |
        echo "OK!" && exit 0
    - path: a_test_resource.txt
      dest: a_test_resource.txt
platforms:
  - type: docker
    image: ubuntu:latest
    setup:
      - type: docker
        run: |
          ls -latr .
    test_setup:
      - type: docker
        copy: ["a_test_resource.txt /opt/a_test_resource.txt"]
      - type: docker
        run: |
          [ -f "/opt/a_test_resource.txt" ] || exit 8
  - type: native
  - type: nextflow
HERE

viash test ./src/dummycomponent/config.vsh.yaml -k true --setup build
```

```
Running tests in temporary directory: '/tmp/viash_test_myscript8420860912742036571'
====================================================================
+/tmp/viash_test_myscript8420860912742036571/build_executable/myscript ---verbosity 6 ---setup build
[notice] Building container 'dummycomponent/myscript:test' with Dockerfile
[info] Running 'docker build -t dummycomponent/myscript:test --no-cache /tmp/viash_test_myscript8420860912742036571/build_executable -f /tmp/viash_test_myscript8420860912742036571/build_executable/tmp/dockerbuild-myscript-cxoQgo/Dockerfile'
#0 building with "default" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 416B done
#1 DONE 0.1s

#2 [internal] load metadata for docker.io/library/ubuntu:latest
#2 DONE 0.8s

#3 [internal] load .dockerignore
#3 transferring context: 2B done
#3 DONE 0.1s

#4 [1/4] FROM docker.io/library/ubuntu:latest@sha256:f9d633ff6640178c2d0525017174a688e2c1aef28f0a0130b26bd5554491f0da
#4 CACHED

#5 [internal] load build context
#5 transferring context: 2B done
#5 DONE 0.1s

#6 [3/4] COPY a_test_resource.txt /opt/a_test_resource.txt
#6 ERROR: failed to calculate checksum of ref 00c8117d-9c86-43b6-878f-e9400a44f2ce::xcdvluk83kvztnda395n9c2j0: "/a_test_resource.txt": not found

#7 [2/4] RUN ls -latr .
#7 0.359 total 17
#7 0.359 drwxr-xr-x   2 root root   2 Apr 18  2022 home
#7 0.359 drwxr-xr-x   2 root root   2 Apr 18  2022 boot
#7 0.359 lrwxrwxrwx   1 root root   8 Feb 12 14:02 sbin -> usr/sbin
#7 0.359 lrwxrwxrwx   1 root root  10 Feb 12 14:02 libx32 -> usr/libx32
#7 0.359 lrwxrwxrwx   1 root root   9 Feb 12 14:02 lib64 -> usr/lib64
#7 0.359 lrwxrwxrwx   1 root root   9 Feb 12 14:02 lib32 -> usr/lib32
#7 0.359 lrwxrwxrwx   1 root root   7 Feb 12 14:02 lib -> usr/lib
#7 0.359 lrwxrwxrwx   1 root root   7 Feb 12 14:02 bin -> usr/bin
#7 0.359 drwxr-xr-x  14 root root  14 Feb 12 14:02 usr
#7 0.359 drwxr-xr-x   2 root root   2 Feb 12 14:02 srv
#7 0.359 drwxr-xr-x   2 root root   2 Feb 12 14:02 opt
#7 0.359 drwxr-xr-x   2 root root   2 Feb 12 14:02 mnt
#7 0.359 drwxr-xr-x   2 root root   2 Feb 12 14:02 media
#7 0.359 drwxr-xr-x  11 root root  13 Feb 12 14:06 var
#7 0.359 drwx------   2 root root   4 Feb 12 14:06 root
#7 0.359 drwxrwxrwt   2 root root   2 Feb 12 14:06 tmp
#7 0.359 drwxr-xr-x   5 root root   5 Feb 12 14:06 run
#7 0.359 drwxr-xr-x  32 root root  73 Feb 12 14:06 etc
#7 0.359 drwxr-xr-x  17 root root  23 Feb 16 09:55 ..
#7 0.359 drwxr-xr-x  17 root root  23 Feb 16 09:55 .
#7 0.359 dr-xr-xr-x  13 root root   0 Feb 19 08:34 sys
#7 0.359 dr-xr-xr-x 741 root root   0 Feb 19 08:34 proc
#7 0.359 drwxr-xr-x   5 root root 340 Feb 19 08:34 dev
#7 DONE 0.4s
------
 > [3/4] COPY a_test_resource.txt /opt/a_test_resource.txt:
------
Dockerfile:8
--------------------
   6 |     RUN ls -latr .
   7 |
   8 | >>> COPY a_test_resource.txt /opt/a_test_resource.txt
   9 |     RUN [ -f "/opt/a_test_resource.txt" ] || exit 8
  10 |
--------------------
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref 00c8117d-9c86-43b6-878f-e9400a44f2ce::xcdvluk83kvztnda395n9c2j0: "/a_test_resource.txt": not found
[error] Error occurred while building container 'dummycomponent/myscript:test'
ERROR! Setup failed!
Unexpected error occurred! If you think this is a bug, please post
create an issue at https://github.com/viash-io/viash/issues containing
a reproducible example and the stack trace below.

viash - 0.8.4
Stacktrace:
java.lang.RuntimeException: Setup failed!
	at io.viash.ViashTest$.apply(ViashTest.scala:134)
	at io.viash.Main$.mainCLI(Main.scala:253)
	at io.viash.Main$.mainCLIOrVersioned(Main.scala:123)
	at io.viash.Main$.main(Main.scala:58)
	at io.viash.Main.main(Main.scala)
```

```
$ tree /tmp/viash_test_myscript8420860912742036571
/tmp/viash_test_myscript8420860912742036571
└── build_executable
    ├── myscript
    ├── tmp
    └── _viash_build_log.txt

2 directories, 2 files
```

**Comment by Grifs** [comment]
Solved in Viash 0.8.5
